### DISCUSSION TITLE
Can my state objects have functions? How about a toJSON function?

### DISCUSSION THREAD

**Discussion by ghost** [pmndrs/valtio#785] [Q&A] [answered]
Hello, I'm trying to keep certain fields from my state objects from being serialized into JSON. Will there be any issues doing this or things to watch out for?

**Comment by dai-shi** [comment, marked as answer]
Yes, but be careful with `this`. If you are not familiar with `this` behavior, better to avoid it.

```js
const state = proxy({
  foo: 'hey',
  toJSON: () => JSON.stringify(state),
});
```

**Reply by ghost** [reply]
Thanks!

**Comment by ghost** [comment]
One note about this if anyone else is looking - Whatever fields you exclude in your `toJSON` function will not be visible in Redux dev tools if you're using that at all.
